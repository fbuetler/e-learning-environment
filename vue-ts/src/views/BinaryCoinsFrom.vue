<template>
  <Game :type="type" :args="args">
    <h2 slot="title">{{ title }}</h2>
    <div slot="description">
      <p>
        In Oberbiberland verwendet man die Bibercoins der folgenden Grösse
      </p>
      <div class="flex-item flex-center flex-row">
        <div><img :src="require('@/assets/coins/binary/sixtyfour.png')" /></div>
        <div><img :src="require('@/assets/coins/binary/thirtytwo.png')" /></div>
        <div><img :src="require('@/assets/coins/binary/sixteen.png')" /></div>
        <div><img :src="require('@/assets/coins/binary/eight.png')" /></div>
        <div><img :src="require('@/assets/coins/binary/four.png')" /></div>
        <div><img :src="require('@/assets/coins/binary/two.png')" /></div>
        <div><img :src="require('@/assets/coins/binary/one.png')" /></div>
      </div>
      <p>
        Alle Bewohner und Bewohnerinnen von Oberbiberland halten sich an die
        binäre Regel. Die binäre Regel bedeutet, dass von jeder Münzgrösse zum
        Zahlen nur eine verwendet werden darf. So brauchen die Biber nicht zu
        viele Münzen bei sich tragen. Welche Summe ergeben die bereitgelegten
        Münzen?
      </p>
    </div>
  </Game>
</template>

<script lang="ts">
import Vue from "vue";
import { Component } from "vue-property-decorator";
import Game, { GameType } from "@/components/Game.vue";
import views from "@/views/Views";
import { coinType } from "@/components/coins/CoinsMixin.vue";

@Component<BinaryCoinsFrom>({
  components: {
    Game,
  },
})
export default class BinaryCoinsFrom extends Vue {
  type: GameType = GameType.COINSFROM;
  title = views.find((el) => el.component === GameType.COINSFROM).title;
  args = { coinType: coinType.BINARY };
  video = "placeholder.mp4";
  description = ``;
}
</script>

<style scoped>
img {
  height: 3em;
}
</style>
