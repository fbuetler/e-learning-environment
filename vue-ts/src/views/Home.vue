<template>
  <div class="cards">
    <div class="card" v-for="item in items" :key="item.id">
      <router-link :to="item.link">
        <img :src="require('@/assets/' + item.img)" style="width:100%" />
        <div class="container">
          {{ item.title }}
        </div>
      </router-link>
    </div>
  </div>
</template>

<script lang="ts">
import Vue from "vue";
import { Component } from "vue-property-decorator";
import { symbolDecryptionTitle } from "./SymbolDecryption.vue";
import { symbolEncryptionTitle } from "./SymbolEncryption.vue";
import { patternDecryptionTitle } from "./PatternDecryption.vue";
import { patternEncryptionTitle } from "./PatternEncryption.vue";

@Component<Home>({})
export default class Home extends Vue {
  private items: Array<{
    id: number;
    title: string;
    link: string;
    img: string;
  }> = [
    {
      id: 1,
      title: "Ähnliche Wörter (hinzufügen)",
      link: "/wordsadd",
      img: "placeholder.jpg",
    },
    {
      id: 2,
      title: "Ähnliche Wörter (ändern)",
      link: "/wordschange",
      img: "placeholder.jpg",
    },
    {
      id: 3,
      title: "Ähnliche Wörter (entfernen)",
      link: "/wordsremove",
      img: "placeholder.jpg",
    },
    {
      id: 4,
      title: "Ähnliche Wörter (vertauschen)",
      link: "/wordsswap",
      img: "placeholder.jpg",
    },
    {
      id: 5,
      title: "Maya Zahlen darstellen",
      link: "/mayasto",
      img: "placeholder.jpg",
    },
    {
      id: 6,
      title: "Maya Zahlen verstehen",
      link: "/mayasfrom",
      img: "placeholder.jpg",
    },
    {
      id: 7,
      title: "Maya Zahlen addieren",
      link: "/mayasaddition",
      img: "placeholder.jpg",
    },
    {
      id: 12,
      title: symbolEncryptionTitle,
      link: "/symbolencryption",
      img: "placeholder.jpg",
    },
    {
      id: 13,
      title: symbolDecryptionTitle,
      link: "/symboldecryption",
      img: "placeholder.jpg",
    },
    {
      id: 14,
      title: patternEncryptionTitle,
      link: "/patternencryption",
      img: "placeholder.jpg",
    },
    {
      id: 15,
      title: patternDecryptionTitle,
      link: "/patterndecryption",
      img: "placeholder.jpg",
    },
    {
      id: 8,
      title: "Reihe aus 3 Bäumen",
      link: "/treerow3",
      img: "placeholder.jpg",
    },
    {
      id: 9,
      title: "3x3 Baumsudoku",
      link: "/treesudoku3",
      img: "placeholder.jpg",
    },
    {
      id: 10,
      title: "Reihe aus 4 Bäumen",
      link: "/treerow4",
      img: "placeholder.jpg",
    },
    {
      id: 11,
      title: "4x4 Baumsudoku",
      link: "/treesudoku4",
      img: "placeholder.jpg",
    },
  ];
}
</script>

<style scoped>
.cards {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(300px, 1fr));
  grid-auto-rows: auto;
  grid-gap: 1rem;
}
.container {
  padding: 2px 16px;
}
/* router-link */
a {
  text-decoration: none;
  color: black;
}
</style>
