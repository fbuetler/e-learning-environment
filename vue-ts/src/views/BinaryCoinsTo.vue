<template>
  <Game :type="type">
    <h2 slot="title">{{ title }}</h2>
  </Game>
</template>

<script lang="ts">
import Vue from "vue";
import { Component } from "vue-property-decorator";
import Game, { GameType } from "../components/Game.vue";
import views from "./Views";

@Component<BinaryCoinsTo>({
  components: {
    Game,
  },
})
export default class BinaryCoinsTo extends Vue {
  type: GameType = GameType.COINSTO;
  title = views.find((el) => el.component === GameType.COINSTO).title;
  video = "placeholder.mp4";
  description = ``;
}
</script>
