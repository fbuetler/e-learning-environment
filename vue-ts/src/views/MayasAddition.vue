<template>
  <Game :type="type" :title="title" />
</template>

<script lang="ts">
import Vue from "vue";
import { Component, Prop } from "vue-property-decorator";
import Game, { GameType } from "../components/Game.vue";
import views from "./Views";

@Component<MayasAddition>({
  components: {
    Game,
  },
})
export default class MayasAddition extends Vue {
  private type: GameType = GameType.MAYASADDITION;
  title = views.find((el) => el.component === this.constructor.name).title;
}
</script>
