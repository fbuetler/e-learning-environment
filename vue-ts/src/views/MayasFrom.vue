<template>
  <Game :type="type" :title="title" />
</template>

<script lang="ts">
import Vue from "vue";
import { Component } from "vue-property-decorator";
import Game, { GameType } from "../components/Game.vue";
import views from "./Views";

@Component<MayasFrom>({
  components: {
    Game,
  },
})
export default class MayasFrom extends Vue {
  private type: GameType = GameType.MAYASFROM;
  title = views.find((el) => el.component === GameType.MAYASFROM).title;
}
</script>
