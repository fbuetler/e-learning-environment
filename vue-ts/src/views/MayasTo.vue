<template>
  <Game :type="type" :title="title" :video="video" :description="description" />
</template>

<script lang="ts">
import Vue from "vue";
import { Component } from "vue-property-decorator";
import Game, { GameType } from "../components/Game.vue";
import views from "./Views";

@Component<MayasTo>({
  components: {
    Game,
  },
})
export default class MayasTo extends Vue {
  type: GameType = GameType.MAYASTO;
  title = views.find((el) => el.component === GameType.MAYASTO).title;
  video = "placeholder.mp4";
  description = `Der Maya‐Stamm der Biber verwendet Nüsse und Hölzchen, 
  um Zahlen darzustellen. In jeder Darstellung dürfen höchstens vier 
  Nüsse vorkommen. Immer wenn fünf Nüsse zusammenkommen, müssen sie 
  gegen ein Hölzchen ausgetauscht werden. Stelle die Zahl in Nüssen 
  und Hölzchen dar.`;
}
</script>
