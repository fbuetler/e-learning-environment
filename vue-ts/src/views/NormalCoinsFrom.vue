<template>
  <Game :type="type" :title="title" />
</template>

<script lang="ts">
import Vue from "vue";
import { Component } from "vue-property-decorator";
import Game, { GameType } from "../components/Game.vue";
import views from "./Views";

@Component<NormalCoinsFrom>({
  components: {
    Game,
  },
})
export default class NormalCoinsFrom extends Vue {
  type: GameType = GameType.COINSFROM;
  title = views.find((el) => el.component === GameType.COINSFROM).title;
  video = "placeholder.mp4";
  description = ``;
}
</script>
