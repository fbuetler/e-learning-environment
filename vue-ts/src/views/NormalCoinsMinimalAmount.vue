<template>
  <Game :type="type" :title="title" :video="video" :description="description" />
</template>

<script lang="ts">
import Vue from "vue";
import { Component } from "vue-property-decorator";
import Game, { GameType } from "../components/Game.vue";
import views from "./Views";

@Component<NormalCoinsMinimalAmount>({
  components: {
    Game,
  },
})
export default class NormalCoinsMinimalAmount extends Vue {
  type: GameType = GameType.COINSMINIMALAMOUNT;
  title = views.find((el) => el.component === GameType.COINSMINIMALAMOUNT)
    .title;
  video = "placeholder.mp4";
  description = "";
}
</script>
