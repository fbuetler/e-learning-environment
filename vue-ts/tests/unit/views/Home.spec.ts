import { mount, RouterLinkStub } from "@vue/test-utils";
import Home from "@/views/Home.vue";
import views from "@/views/Views";

expect.extend({
  toBeDistinct(received) {
    if (Array.isArray(received) && new Set(received).size === received.length) {
      return {
        message: () => `expected [${received}] has distinct values`,
        pass: true,
      };
    } else {
      return {
        message: () => `expected [${received}] has not distinct values`,
        pass: false,
      };
    }
  },
});

describe("Home.vue", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(Home, {
      stubs: {
        RouterLink: RouterLinkStub,
      },
    });
  });

  it("is a Vue instance", () => {
    expect(wrapper.vm).toBeTruthy();
  });

  views.forEach((view) => {
    it(`is view '${view.title}' present`, () => {
      const routerLinks = wrapper
        .findAllComponents(RouterLinkStub)
        .filter((w) => w.props().to === view.path);
      expect(routerLinks.length).toBe(1);
      const routerLink = routerLinks.at(0);
      expect(routerLink.exists()).toBe(true);
      expect(routerLink.html()).toContain(view.title);
      expect(routerLink.find("img").exists()).toBe(true);
    });
  });

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
