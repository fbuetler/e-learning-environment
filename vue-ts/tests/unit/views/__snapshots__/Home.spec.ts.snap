// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue renders correctly 1`] = `
<div
  class="cards"
>
  <div
    class="card clickable"
  >
    <a>
      <div
        class="card-image flex-item flex-center"
      >
        <img
          src=""
        />
      </div>
       
      <div
        class="card-container"
      >
        
        Ähnliche Wörter - Buchstabe hinzufügen
      
      </div>
    </a>
  </div>
  <div
    class="card clickable"
  >
    <a>
      <div
        class="card-image flex-item flex-center"
      >
        <img
          src=""
        />
      </div>
       
      <div
        class="card-container"
      >
        
        Ähnliche Wörter - Buchstabe austauschen
      
      </div>
    </a>
  </div>
  <div
    class="card clickable"
  >
    <a>
      <div
        class="card-image flex-item flex-center"
      >
        <img
          src=""
        />
      </div>
       
      <div
        class="card-container"
      >
        
        Ähnliche Wörter - Buchstabe entfernen
      
      </div>
    </a>
  </div>
  <div
    class="card clickable"
  >
    <a>
      <div
        class="card-image flex-item flex-center"
      >
        <img
          src=""
        />
      </div>
       
      <div
        class="card-container"
      >
        
        Ähnliche Wörter - Buchstaben vertauschen
      
      </div>
    </a>
  </div>
  <div
    class="card clickable"
  >
    <a>
      <div
        class="card-image flex-item flex-center"
      >
        <img
          src=""
        />
      </div>
       
      <div
        class="card-container"
      >
        
        Maya Zahlen darstellen
      
      </div>
    </a>
  </div>
  <div
    class="card clickable"
  >
    <a>
      <div
        class="card-image flex-item flex-center"
      >
        <img
          src=""
        />
      </div>
       
      <div
        class="card-container"
      >
        
        Maya Zahlen verstehen
      
      </div>
    </a>
  </div>
  <div
    class="card clickable"
  >
    <a>
      <div
        class="card-image flex-item flex-center"
      >
        <img
          src=""
        />
      </div>
       
      <div
        class="card-container"
      >
        
        Maya Zahlen addieren
      
      </div>
    </a>
  </div>
  <div
    class="card clickable"
  >
    <a>
      <div
        class="card-image flex-item flex-center"
      >
        <img
          src=""
        />
      </div>
       
      <div
        class="card-container"
      >
        
        Zahlen mit Münzen und Geldscheinen
      
      </div>
    </a>
  </div>
  <div
    class="card clickable"
  >
    <a>
      <div
        class="card-image flex-item flex-center"
      >
        <img
          src=""
        />
      </div>
       
      <div
        class="card-container"
      >
        
        Münzbeträge ausrechnen
      
      </div>
    </a>
  </div>
  <div
    class="card clickable"
  >
    <a>
      <div
        class="card-image flex-item flex-center"
      >
        <img
          src=""
        />
      </div>
       
      <div
        class="card-container"
      >
        
        Mit möglichst wenig Münzen bezahlen
      
      </div>
    </a>
  </div>
  <div
    class="card clickable"
  >
    <a>
      <div
        class="card-image flex-item flex-center"
      >
        <img
          src=""
        />
      </div>
       
      <div
        class="card-container"
      >
        
        Gleichwertige Münzen tauschen
      
      </div>
    </a>
  </div>
  <div
    class="card clickable"
  >
    <a>
      <div
        class="card-image flex-item flex-center"
      >
        <img
          src=""
        />
      </div>
       
      <div
        class="card-container"
      >
        
        Bezahlen mit binären Münzen
      
      </div>
    </a>
  </div>
  <div
    class="card clickable"
  >
    <a>
      <div
        class="card-image flex-item flex-center"
      >
        <img
          src=""
        />
      </div>
       
      <div
        class="card-container"
      >
        
        Binäre Münzbeträge ausrechnen
      
      </div>
    </a>
  </div>
  <div
    class="card clickable"
  >
    <a>
      <div
        class="card-image flex-item flex-center"
      >
        <img
          src=""
        />
      </div>
       
      <div
        class="card-container"
      >
        
        mit Muster dechiffrieren
      
      </div>
    </a>
  </div>
  <div
    class="card clickable"
  >
    <a>
      <div
        class="card-image flex-item flex-center"
      >
        <img
          src=""
        />
      </div>
       
      <div
        class="card-container"
      >
        
        mit Muster chiffrieren
      
      </div>
    </a>
  </div>
  <div
    class="card clickable"
  >
    <a>
      <div
        class="card-image flex-item flex-center"
      >
        <img
          src=""
        />
      </div>
       
      <div
        class="card-container"
      >
        
        mit Zeichen dechiffrieren
      
      </div>
    </a>
  </div>
  <div
    class="card clickable"
  >
    <a>
      <div
        class="card-image flex-item flex-center"
      >
        <img
          src=""
        />
      </div>
       
      <div
        class="card-container"
      >
        
        mit Zeichen chiffrieren
      
      </div>
    </a>
  </div>
  <div
    class="card clickable"
  >
    <a>
      <div
        class="card-image flex-item flex-center"
      >
        <img
          src=""
        />
      </div>
       
      <div
        class="card-container"
      >
        
        Reihe aus 3 Bäumen
      
      </div>
    </a>
  </div>
  <div
    class="card clickable"
  >
    <a>
      <div
        class="card-image flex-item flex-center"
      >
        <img
          src=""
        />
      </div>
       
      <div
        class="card-container"
      >
        
        3x3 Baumsudoku
      
      </div>
    </a>
  </div>
  <div
    class="card clickable"
  >
    <a>
      <div
        class="card-image flex-item flex-center"
      >
        <img
          src=""
        />
      </div>
       
      <div
        class="card-container"
      >
        
        Reihe aus 4 Bäumen
      
      </div>
    </a>
  </div>
  <div
    class="card clickable"
  >
    <a>
      <div
        class="card-image flex-item flex-center"
      >
        <img
          src=""
        />
      </div>
       
      <div
        class="card-container"
      >
        
        4x4 Baumsudoku
      
      </div>
    </a>
  </div>
</div>
`;
